### PR TITLE
[4844] [Hive] Fix fcuV3 parameter return 

### PR DESCRIPTION
**Pull request by Gabriel-Trintinalia**
## PR description
Fix some hive tests

- Return JsonRpcErrorResponse instead of JsonRpcSuccessResponse when fork not supported or parameters invalid.
- Validate parameter before fork validation
- Validate getParentBeaconBlockRoot on EngineForkchoiceUpdatedV3

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineForkchoiceUpdatedV3.java [hunk 1]
```diff
@@ -48,14 +49,21 @@ public String getName() {
...
+      final EngineForkchoiceUpdatedParameter fcuParameter,
+      final Optional<EnginePayloadAttributesParameter> maybePayloadAttributes) {
     if (fcuParameter.getHeadBlockHash() == null) {
       return ValidationResult.invalid(RpcErrorType.INVALID_PARAMS, "Missing head block hash");
     } else if (fcuParameter.getSafeBlockHash() == null) {
       return ValidationResult.invalid(RpcErrorType.INVALID_PARAMS, "Missing safe block hash");
     } else if (fcuParameter.getFinalizedBlockHash() == null) {
       return ValidationResult.invalid(RpcErrorType.INVALID_PARAMS, "Missing finalized block hash");
     }
+    if (maybePayloadAttributes.isPresent()) {
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineForkchoiceUpdatedV3.java:61]: this is the parameter checking you mean that will potentially conflict with the larger refactor?

**Review by macfarla — APPROVED** [review]
LGTM

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineForkchoiceUpdatedV3.java (hunk 1 above)
Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineForkchoiceUpdatedV3.java:61]: I think fine to fix it like this in the short-term to clear the noise for the hive tests

Comment (Gabriel-Trintinalia) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineForkchoiceUpdatedV3.java:61]: We would not necessarily have to validate this if the plans are to use strict JSON validation

**Pull request merged**